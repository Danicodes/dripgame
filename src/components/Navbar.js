import { NavLink } from 'react-router-dom';

export default function Navbar(){
    return (
        <nav className='navbar'>
            <ul>
                <li>
                    <NavLink exact to='/' activeClassName='activeNav'>Home</NavLink>
                </li>
                <li>
                    <NavLink exact to='/about' activeClassName='activeNav'>About</NavLink>
                </li>
                <li>
                    <NavLink exact to='/contribute' activeClassName='activeNav'>Contribute</NavLink>
                </li>
                <li>
                    <NavLink exact to='/artwork' activeClassName='activeNav'>Artwork</NavLink>
                </li>
            </ul>
        </nav>
    );
}