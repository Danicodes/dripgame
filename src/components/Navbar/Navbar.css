/* Currently active navigation tab */
li>.active {
    color:rgb(106, 67, 203);
    font-weight: lighter;
}

.navbar {
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    font-size: 1.2rem;
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.navbar li {
    list-style-type: none;
    float: left;
    text-decoration: none;
    
}

.navbar li:hover {
    background-color: rgba(57, 77, 124, 0.506);
}
.navbar>ul {
    display: grid;
    grid-template-columns: repeat(5, auto);
    grid-gap: 2px;
    list-style: none;
    text-align: center;
    width: 70vw;
    margin-right: 2rem;
}

