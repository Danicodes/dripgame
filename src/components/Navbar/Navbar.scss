/* Currently active navigation tab */
@use '../../style' as *;

.navbar {
    font-family: $normal-font;
    font-size: $medium-font-size;
    height: 50px;
    display: inline;
    justify-content: center;
    align-items: center;

    .navbar-links {
        background-color: $medium-color;
        width: fit-content;
        height: 10vh;
    }

    li {
        list-style-type: none;
        float: left;

        a {
           display: block;
           text-decoration-line: none;
           padding: $padding-top $padding-left;
           font-size: $medium-font-size;
           color: $font-color;
        }
    }

    a.active {
        background-color: rgba($secondary-bg-color, 0.5);
        color: rgba($font-color, 0.7);
        font-weight: lighter;
    }

    li:hover {
        background-color: rgba($medium-color, 0.5);
    }
    >ul {
        display: grid;
        grid-template-columns: repeat(5, auto);
        list-style: none;
        text-align: center;
        width: 80vw;
    }

    border-bottom: 5px solid rgba(23, 2, 37, 0.865);
}

