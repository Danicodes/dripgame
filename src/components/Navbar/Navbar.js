import { NavLink } from 'react-router-dom';
import './Navbar.css';

export default function Navbar(){
    return (
        <nav className='navbar'>
            <ul>
                <li>
                    <NavLink exact to='/' activeClassName='activeNav'>Home</NavLink>
                </li>
                <li>
                    <NavLink exact to='/about' activeClassName='activeNav'>About</NavLink>
                </li>
                <li>
                    <NavLink exact to='/contribute' activeClassName='activeNav'>Contribute</NavLink>
                </li>
                <li>
                    <NavLink exact to='/artwork' activeClassName='activeNav'>Artwork</NavLink>
                </li>  
                <li>
                    <NavLink exact to='/admin' activeClassName='activeNav'>Admin</NavLink>
                </li>
            </ul>
        </nav>
    );
}