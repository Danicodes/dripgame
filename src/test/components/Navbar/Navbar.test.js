import { render, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom'
import '@testing-library/jest-dom'

import Navbar from '../../../components/Navbar/Navbar';

describe('the navbar', () => {

    test("it renders the navbar component", () => {
        const tree = render(<Navbar />, { wrapper: BrowserRouter });
        expect(tree).toMatchSnapshot();
    })

    test("only current page is active", () => {
        render(<Navbar />, { wrapper: BrowserRouter });
        expect(screen.getByText('Home')).toHaveClass('active')
        expect(screen.getByText('About')).not.toHaveClass('active')
    })
})