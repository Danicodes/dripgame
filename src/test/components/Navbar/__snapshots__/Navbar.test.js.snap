// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the navbar it renders the navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="logoDiv"
        >
          <img
            class="logoImage"
            src="logo192.png"
          />
          <a
            class="logoLink"
            href="#"
          >
            <p
              class="logoTitle"
            >
              DripGame
            </p>
          </a>
        </div>
        <nav
          class="navbar"
        >
          <ul
            class="navbar-links"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class=""
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                class=""
                href="/contribute"
              >
                Contribute
              </a>
            </li>
            <li>
              <a
                class=""
                href="/artwork"
              >
                Artwork
              </a>
            </li>
            <li>
              <a
                class=""
                href="/admin"
              >
                Admin
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="logoDiv"
      >
        <img
          class="logoImage"
          src="logo192.png"
        />
        <a
          class="logoLink"
          href="#"
        >
          <p
            class="logoTitle"
          >
            DripGame
          </p>
        </a>
      </div>
      <nav
        class="navbar"
      >
        <ul
          class="navbar-links"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class=""
              href="/contribute"
            >
              Contribute
            </a>
          </li>
          <li>
            <a
              class=""
              href="/artwork"
            >
              Artwork
            </a>
          </li>
          <li>
            <a
              class=""
              href="/admin"
            >
              Admin
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
